core/src/main/java/com/testaem/aem/core/services/impl/RssNewsResourceBuilderService.java

```
@@ -0,0 +1,110 @@
+    private static final String LAYOUT_CONFIG = "/conf/testaem/settings/wcm/templates/narutopage";
+    private static final Logger LOGGER = LoggerFactory.getLogger(RssNewsResourceBuilderService.class);
+    @Reference
+    private Supplier<ResourceResolver> sysUserResourceResolverProvider;
+    private final ResourceResolver resourceResolver = sysUserResourceResolverProvider.get();
+
+
+    @Override
+    public List<Resource> build(Feed feed) {
+
+        Resource aggregatorPage = resourceResolver.getResource(PAGES_FOLDER);
+
+        if (Objects.isNull(aggregatorPage)) {
+            return null;
+        }
+
+        List<FeedMessage> hotNews = feed.getMessages()
+                .stream()
+                .filter(item -> Objects.isNull(aggregatorPage.getChild(item.getGuid())))
+                .collect(Collectors.toList());
+
+
+        return hotNews.stream()
+                .map(news -> createPage(news, aggregatorPage))
+                .collect(Collectors.toList());
+    }
+
+    private Resource createPage(FeedMessage message, Resource parentResource) {
+        Resource contentPage;
+        LOGGER.info("Creating page with guid: " + message.getGuid());
+        try {
+            Resource page = resourceResolver.create(parentResource, message.getGuid(), prepareParamMap(Type.PAGE, message));
+            Resource content = resourceResolver.create(page, JcrConstants.JCR_CONTENT, prepareParamMap(Type.CONTENT, message));
+            Resource root = resourceResolver.create(content, "root", prepareParamMap(Type.CONTAINER_NODE, message));
+            Resource container = resourceResolver.create(root, "container", prepareParamMap(Type.CONTAINER_NODE, message));
+            contentPage = resourceResolver.create(container, "contentpage", prepareParamMap(Type.CONTENT_PAGE, message));
+            LOGGER.info("page with id = " + message.getGuid() + " has been created");
+        } catch (PersistenceException e) {
+            throw new RuntimeException(e);
```

(important) Bad practice here. You are technically wrapping a more "severe" exception into a less severe one, half-swallowing it. You just lose the chance to handle exceptions properly.

Every kind of exceptions you can foresee, must be properly logged

---

log.error added before throwing runtime